### ISSUE TITLE
Reproducing LSTM Checkpoint

### ISSUE THREAD

**Issue by nicholasshort** [prajwalsingh/EEG2Image#11]
Hello,

I have attempted to retrain the LSTM on the provided dataset from scratch. After a few hundred epochs, I haven't seen any improvements. I am running train.py under lstm_kmean. I was wondering if this is the final training code you used to train your LSTM. Is there something else I should try? Any help would be greatly appreciated.

Thanks!

**Comment by nicholasshort** [comment]
Never mind! Got it to train, thanks

**Comment by prajwalsingh** [comment]
> Never mind! Got it to train, thanks

Hi @nicholasshort, good to know it works. May I know what exactly the problem was?

**Comment by Akshaykushawaha** [comment]
Hey, @nicholasshort @prajwalsingh !
I was trying to get the model to run with the same data and code as provided, but I am facing the issue just like Nicholas got, of not getting any improvement over the epochs; the loss mostly remains above 99%, and the final accuracy fails to surpass 15%.
It would help if you could guide me on how you resolved the issue since I have been trying to get the code to run for a very long time now, but all in vain.
I appreciate any help you can provide.
Below is the resulting image I am getting.
![image](https://github.com/prajwalsingh/EEG2Image/assets/69956089/b8f73c55-e229-47dc-aad3-606803031c92)

**Comment by prajwalsingh** [comment]
Hi @Akshaykushawaha , can you tell what exactly loss above 99% implies? Is it training accuracy or something else? Cross check if dataset you are loading is correct or not and while inference you are loading correct class data.

**Comment by Akshaykushawaha** [comment]
Thanks for the quick response, this is the shape of the data that is being input (used the one you have provided) :
![image](https://github.com/prajwalsingh/EEG2Image/assets/69956089/226d8ed8-b943-400b-a11d-f29b8f2a5553)

while training i get this kind of loss till 3000th epoch and the accuracy doesn't get any improvement:
Train Epoch: 422, Loss: 0.9916849136352539: 100%|██████████| 177/177 [00:10<00:00, 16.21it/s]

Test Epoch: 422, Loss: 0.9900215864181519: 100%|██████████| 23/23 [00:01<00:00, 18.05it/s]

**Comment by prajwalsingh** [comment]
@Akshaykushawaha , currently travelling. Will check this as soon as possible. I am not able to recall if this much accuracy I was getting on test data. Something is not correct.

**Comment by Akshaykushawaha** [comment]
Sure thing!👍

**Comment by Akshaykushawaha** [comment]
Hey @prajwalsingh! do I open a new issue since the last one was closed without the solution?

**Comment by Akshaykushawaha** [comment]
@prajwalsingh @nicholasshort are there any updates regarding this? I am still stuck on this issue..

**Comment by NZW666666** [comment]
您好，我也正在考虑这个问题，求更新

**Comment by prajwalsingh** [comment]
Hi @Akshaykushawaha and @NZW666666 ,

I have updated the code of eeg feature extraction for better understanding. 

1. First, you have to download the thought viz dataset pickle file and its equivalent images (images are not required for feature extraction learning, just class names are required).
2. Then run train.py code and it will run for around 3000 epochs.
3. After train, now run the find_bestckpy.py code, it will give you the path of best checkpoint with better test acc.
4. Finally you can run inference.py for generating tsne plot.

PFA the snapshot for the same, I have re-run the code and got the reported ACC as per the paper.

为了更好地理解，我更新了脑电图特征提取的代码。

1.  首先，您必须下载思想可视化数据集 pickle 文件及其等效图像（特征提取学习不需要图像，只需要类名称）。
2.  然后运行 train.py 代码，它将运行大约 3000 个 epoch。
3.  训练结束后，现在运行 find_bestckpy.py 代码，它将为您提供具有更好测试精度的最佳检查点的路径。
4.  最后，您可以运行 inference.py 来生成 tsne 图。

PFA 的快照相同，我重新运行了代码并根据论文获取了报告的 ACC。

![eeg2image_rerun](https://github.com/prajwalsingh/EEG2Image/assets/22240769/2222ac80-4203-4724-8004-efef2b77adf4)

![embedding](https://github.com/prajwalsingh/EEG2Image/assets/22240769/3f818e52-4b96-4670-a547-ffa681903a1b)

**Comment by Akshaykushawaha** [comment]
@prajwalsingh Hey man! its working, thanks a lot for the resolution!!
